pkg/marshal/marshal.go

```diff
@@ -21,6 +21,13 @@ type RawQuery[I any] struct {
 	Detail string `json:"detail"`
 }
 
+type RawQueryRaw[I any] struct {
```

Comment:
Maybe we can call it RawQuerySingle or something along those lines or we can change the original RawQuery to me indeed of an array of I and then refactor according to that

Reply:
original RawQuery only handles when `result` field is array, but `result` field can be an object or basic types, golang doesn't allow `any` to be casted to `[]any` and vice versa, that's why a new struct and a new funciton are created.

Reply:
Yeah, you're right my bad.